src/convert_hl7v2_fhir/controllers/hl7/hl7_message_controller.py

```diff
@@ -0,0 +1,262 @@
+        self.metadata = metadata or {
+            "organization": {
+                "identifier": {"value": "XXX"},
+                "name": "SIMULATED HOSPITAL NHS FOUNDATION TRUST",
+            },
+            "location": {
+                "identifier": {"value": "XXXY1"},
+                "address": {"postalCode": "XX20 5XX", "city": "Exampletown"},
+            },
+        }
+
+    def to_fhir_bundle(self) -> Bundle:
+        bundle_json = {
+            "resourceType": "Bundle",
+            "type": "message",
+            "meta": {
+                "profile": [
+                    "https://fhir.simplifier.net/Hospital-Activity-Notification-Service/StructureDefinition/ActivityNotification-Bundle"
+                ]
+            },
+            "entry": [
+                self._create_header(),
+                self._create_patient(),
+                self._create_location(),
+                self._create_organization(),
+                self._create_encounter(),
+            ],
+        }
+        return Bundle(**bundle_json)
+
+    def _extract_field(self, segment_name: str, *indexes: int) -> str:
+        try:
+            _field = self.hl7_message[segment_name]
+            for index in indexes:
+                _field = _field[index]
+        except KeyError:
+            raise MissingSegmentError(f"Required segment '{segment_name}' was missing.")
+        except IndexError:
+            field_indexes = ".".join(str(i) for i in indexes)
+            raise MissingFieldOrComponentError(
+                f"Required field '{segment_name}.{field_indexes}' was missing."
+            )
+        return str(_field)
+
+    def _create_header(self) -> Dict[str, Any]:
+        return {
+            "fullUrl": f"urn:uuid:{self.message_header_uuid}",
+            "resource": {
+                "resourceType": "MessageHeader",
+                "id": str(self.message_header_uuid),
+                "meta": {
+                    "profile": [
+                        "https://fhir.simplifier.net/Hospital-Activity-Notification-Service/StructureDefinition/ActivityNotification-UKCore-MessageHeader"
+                    ]
+                },
+                "eventCoding": {
+                    "system": "http://terminology.hl7.org/CodeSystem/v2-0003",
+                    "code": self._extract_field("EVN", 0, 1),
+                },
+                "source": {"endpoint": "http://example.com/fhir/R4"},
+                "responsible": {"reference": f"urn:uuid:{self.organization_uuid}"},
+                "focus": [{"reference": f"urn:uuid:{self.encounter_uuid}"}],
+            },
+        }
+
+    def _create_patient(self) -> Dict[str, Any]:
+        nhs_number = self._extract_nhs_number()
+        family_name = self._extract_field("PID", 0, 5, 0, 0)
+        given_name = [
+            name
+            for name in (
+                self._extract_field("PID", 0, 5, 0, 1),
+                self._extract_field("PID", 0, 5, 0, 2),

```

Comment:
Possibly stupid question, but do we know we're only getting two given names?

Reply:
Ah that's a good spot - we need to space separate the result of `_extract_field("PID", 0, 5, 0, 2)` (if it exists)

These could be space separated according to [the spec on Caristix](https://hl7-definition.caristix.com/v2/HL7v2.8/Fields/PID.5.3)